src/compiler/watchPublic.ts

```diff
@@ -254,6 +254,9 @@ export interface ProgramHost<T extends BuilderProgram> {
      * Returns the module resolution cache used by a provided `resolveModuleNames` implementation so that any non-name module resolution operations (eg, package.json lookup) can reuse it
      */
     getModuleResolutionCache?(): ModuleResolutionCache | undefined;
+
+    /** True if it's safe for the parser to skip parsing JSDoc tags. */
+    skipJSDocParsing?: boolean;

```

Comment:
Is this needed for projects like tsdoc?

Reply:
It's disabled by default anywhere except `tsc` and in some tests; theoretically a downstream user could enable this if they didn't care about the insides of a JSDoc comment, e.g. a formatter like prettier (though I'm not sure if they do or don't reformat JSDoc comments). Of course, prettier is using createSourceFile directly so they wouldn't use this flag, but I can imagine a bundler using this or something.

Reply:
Oh, I forgot that TS Server always needs to have JSDoc parsed for editor features.

Is it easy enough to plumb this option down from the `Project` class? We should enable this property for AutoImportProviderProjects and AuxiliaryProjects since they don’t interact directly with the editor.

Reply:
Oh that's a good idea; I guess I assumed that those would share SourceFiles but I guess no. Let me check and see!

Reply:
Ohh, they might share source files

Reply:
Dang :(